### PR TITLE
Add base template files for GC Infra CI tests in Azure Pipelines

### PR DESCRIPTION
**Pull request by ivdiazsa**
This PR adds a couple template `yml` files to describe Azure Pipelines jobs for GC Benchmarking Infra, as well as a quick infra setup Python script.

### CODE REVIEW

**Comment by ivdiazsa** [comment]
Here's the PR with the initial templates @DrewScoggins

File: azure-pipelines-gc.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  containers:
+    - container: ubuntu_x64_build_container
+      image: microsoft/dotnet-buildtools-prereqs:ubuntu-16.04-c103199-20180628134544
+
+# Trigger builds for PR's targeting master
+pr:
+  branches:
+    include:
+    - master
+  paths:
```
Comment (DrewScoggins) [inline azure-pipelines-gc.yml:11]: Should probably add some include paths here for just the GC area

Comment (ivdiazsa) [reply azure-pipelines-gc.yml:11]: Done! Would this be enough? I'm setting it to trigger anytime the main script, or any source files are modified.

**Review by DrewScoggins — APPROVED** [review]
LGTM

**Pull request merged**